From 5fded890b2ffe5abbe8d7d421f54a67369564d94 Mon Sep 17 00:00:00 2001
From: Beleswar Padhi <b-padhi@ti.com>
Date: Tue, 18 Mar 2025 15:50:08 +0530
Subject: [PATCH 4/5] remoteproc: k3-m4: Prevent TI-SCI messaging

Currently, Device Manager (DM) is not up for tda5-vdk platform.
Therefore, prevent any TI-SCI messaging services in the K3 M4 remoteproc
driver. Harcode the device state to "RPROC_DETACHED" to let it boot in
IPC-only mode currently.

Signed-off-by: Beleswar Padhi <b-padhi@ti.com>
---
 drivers/remoteproc/ti_k3_m4_remoteproc.c | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/drivers/remoteproc/ti_k3_m4_remoteproc.c b/drivers/remoteproc/ti_k3_m4_remoteproc.c
index bfb28ee68231..8e230e8abe99 100644
--- a/drivers/remoteproc/ti_k3_m4_remoteproc.c
+++ b/drivers/remoteproc/ti_k3_m4_remoteproc.c
@@ -44,14 +44,6 @@ static int k3_m4_rproc_probe(struct platform_device *pdev)
 	bool p_state = false;
 	int ret;
 
-	data = of_device_get_match_data(dev);
-	if (!data)
-		return -ENODEV;
-
-	ret = rproc_of_parse_firmware(dev, 0, &fw_name);
-	if (ret)
-		return dev_err_probe(dev, ret, "failed to parse firmware-name property\n");
-
 	rproc = devm_rproc_alloc(dev, dev_name(dev), &k3_m4_rproc_ops, fw_name,
 				 sizeof(*kproc));
 	if (!rproc)
@@ -65,6 +57,7 @@ static int k3_m4_rproc_probe(struct platform_device *pdev)
 	kproc->rproc = rproc;
 	platform_set_drvdata(pdev, rproc);
 
+	/*
 	kproc->ti_sci = devm_ti_sci_get_by_phandle(dev, "ti,sci");
 	if (IS_ERR(kproc->ti_sci))
 		return dev_err_probe(dev, PTR_ERR(kproc->ti_sci),
@@ -95,18 +88,23 @@ static int k3_m4_rproc_probe(struct platform_device *pdev)
 	ret = k3_rproc_of_get_memories(pdev, kproc);
 	if (ret)
 		return ret;
+	*/
 
 	ret = k3_reserved_mem_init(kproc);
 	if (ret)
 		return dev_err_probe(dev, ret, "reserved memory init failed\n");
 
+	/*
 	ret = kproc->ti_sci->ops.dev_ops.is_on(kproc->ti_sci, kproc->ti_sci_id,
 					       &r_state, &p_state);
 	if (ret)
 		return dev_err_probe(dev, ret,
 				     "failed to get initial state, mode cannot be determined\n");
+	*/
+
+	/* Hard-code to boot in IPC-only mode for now */
+	p_state = true;
 
-	/* configure devices for either remoteproc or IPC-only mode */
 	if (p_state) {
 		rproc->state = RPROC_DETACHED;
 		dev_info(dev, "configured M4F for IPC-only mode\n");
-- 
2.34.1

